airflow/utils/dag_cleaner.py

```diff
@@ -0,0 +1,288 @@
+# Licensed to the Apache Software Foundation (ASF) under one
```

Comment:
Unsure about the name "dag_cleaner", it doesn't ring a bell with me. What do you think of:
- dag_clearer
- task_clearer
- task_instance_clearer
- clear_tasks
- clear_task_instances

Reply:
Maybe `dag_utils` in case we will have more things to put there?

Reply:
I'm afraid we don't want to see more things in this module. This leads to further cyclical imports.

Reply:
Small modules with one responsibility are a better idea. 
https://medium.com/@severinperez/writing-flexible-code-with-the-single-responsibility-principle-b71c4f3f883f
In Python, you can look at modules the same way you look at classes. This is very often interchangeable.